### DISCUSSION TITLE
Cmake?

### DISCUSSION THREAD

**Discussion by tksharpless** [micans/mcl#14] [Ideas]
Is there any possibility of adding Cmake scripts, to enable building these tools on platforms that do not support AutoMake?
Yes, I mean Windows, but there are other potential targets.  Moreover, this could ease all future development, as the ancient Gnu tools are so much harder to understand and use.

**Comment by micans** [comment]
Hi, the dependency of the source code goes beyond the build system - this code uses unix system calls for catching signals, pthreads, timing and so on. I have in the distant past compiled mcl on Cygwin and this worked fine; from recent issues in this repository it can be inferred that this still generally works. Would Cygwin be an option for you?

Note that normally for compiling mcl you do not need to have any knowledge about autotools, nor do you need to have them installed. You do need a unix-like build environment, and that is what Cygwin provides.

A further note - sometimes it catches people out if they clone this repository - this is not how to build mcl. For building mcl you normally download tarballs and/or follow the installation instructions on the main mcl GitHub page.

**Comment by tksharpless** [comment]
Thanks, understood.  I develop cross platform software on Windows by
preference, have built Unix software on Cygwin in the past, would do so
again if I had to; but only as a last resort.  Just now I only want to use
MCL to test whether it solves a certain problem well enough to warrant my
developing C++ code for a similar algorithm.   Working up a Windows build
would be wasted effort, as I can easily do those tests on Linux.
